mo_gymnasium/envs/mountain_car/mountain_car.py

```diff
@@ -39,11 +63,19 @@ def step(self, action: int):
             velocity = 0
 
         terminated = bool(position >= self.goal_position and velocity >= self.goal_velocity)
-        # reward = -1.0
-        reward = np.zeros(3, dtype=np.float32)
+
+        reward = np.zeros(self.reward_dim, dtype=np.float32)
+
         reward[0] = 0.0 if terminated else -1.0  # time penalty
-        reward[1] = 0.0 if action != 0 else -1.0  # reverse penalty
-        reward[2] = 0.0 if action != 2 else -1.0  # forward penalty
+
+        if self.merge_move_penalty:
+            reward[1] = 0.0 if action == 1 else -1.0
+        else:
+            reward[1] = 0.0 if action != 0 else -1.0  # reverse penalty
+            reward[2] = 0.0 if action != 2 else -1.0  # forward penalty
+
+        if self.add_speed_objective:
+            reward[-1] = 15 * abs(velocity)
```

Comment:
Why the 15 * scale?

Reply:
@pranavg23

Reply:
Velocity ranges from -0.07 to 0.07, and we are providing the absolute value of that as a reward to ensure that the agent is able to take into consideration the magnitude. Though, in comparison to the time and movement rewards (-1), the velocity reward's magnitude is quite small (max of 0.07), therefore a constant of 15 is multiplied making the maximum around 1.

Reply:
I see, I've been questioning myself about these kinds of scaling in MORL as it is arguably one of the advantages compared to single objective RL: the algorithms should be able to deal with these kinds of scale differences, and thus it's less reward engineering work for humans. 

Yet I understand it makes the problem easier.

I don't have any strong reason to push for removing this but that's just food for thought in MORL. If you have an opinion on this I'm happy to hear it. Otherwise we can merge.

Reply:
I had experimented with no scaling and different magnitudes of scaling. My research questions were along the lines of how effective reward shaping is and analysing the behaviour of the agent with different priorities assigned to the objectives/reward signals. I found out that having comparable reward ranges for the objectives, resulted in diverse solutions (which can help in the testing of an algorithm) and solutions which were unseen in the problem. Therefore, I decided to keep it.